### PR TITLE
Add IIO support for FMCOMMS5.

### PR DESCRIPTION
**Pull request by danmois**
Add IIO support for FMCOMMS5.

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 1]
```diff
@@ -81,6 +81,38 @@ int32_t axi_adc_write(struct axi_adc *adc,
...
+ * @param adc - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - Data value to be written.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int32_t axi_slave_adc_write(struct axi_adc *adc,
+			    uint32_t reg_addr,
+			    uint32_t reg_data)
+{
+	no_os_axi_io_write(adc->slave_base, reg_addr, reg_data);
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:111]: return directly.

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 2]
```diff
@@ -81,6 +81,38 @@ int32_t axi_adc_write(struct axi_adc *adc,
...
+ * @param adc - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - Read data value.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int32_t axi_slave_adc_read(struct axi_adc *adc,
+			   uint32_t reg_addr,
+			   uint32_t *reg_data)
+{
+	no_os_axi_io_read(adc->slave_base, reg_addr, reg_data);
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:95]: return directly.

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 3]
```diff
@@ -523,12 +555,22 @@ int32_t axi_adc_update_active_channels(struct axi_adc *adc, uint32_t mask)
 
 	adc->mask = mask;
 	for (ch = 0; ch < adc->num_channels; ch++) {
-		axi_adc_read(adc, AXI_ADC_REG_CHAN_CNTRL(ch), &val);
+		if (ch > (adc->num_slave_channels - 1))
+			axi_slave_adc_read(adc, AXI_ADC_REG_CHAN_CNTRL(ch - adc->num_slave_channels),
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:559]: handle return

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 4]
```diff
@@ -523,12 +555,22 @@ int32_t axi_adc_update_active_channels(struct axi_adc *adc, uint32_t mask)
...
+					   &val);
+		else
+			axi_adc_read(adc, AXI_ADC_REG_CHAN_CNTRL(ch), &val);
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:562]: ditto.

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 5]
```diff
@@ -523,12 +555,22 @@ int32_t axi_adc_update_active_channels(struct axi_adc *adc, uint32_t mask)
...
+
 		new_val = val & (~AXI_ADC_ENABLE);
 		if (mask & 1)
 			new_val = val | AXI_ADC_ENABLE;
-		if (new_val != val)
-			axi_adc_write(adc, AXI_ADC_REG_CHAN_CNTRL(ch), new_val);
+		if (new_val != val) {
+			if (ch > (adc->num_slave_channels - 1))
+				axi_slave_adc_write(adc, AXI_ADC_REG_CHAN_CNTRL(ch - adc->num_slave_channels),
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:569]: ditto.

File: drivers/axi_core/axi_adc_core/axi_adc_core.c [hunk 6]
```diff
@@ -523,12 +555,22 @@ int32_t axi_adc_update_active_channels(struct axi_adc *adc, uint32_t mask)
...
+						    new_val);
+			else
+				axi_adc_write(adc, AXI_ADC_REG_CHAN_CNTRL(ch), new_val);
```
Comment (amiclaus) [inline drivers/axi_core/axi_adc_core/axi_adc_core.c:572]: ditto.

**Comment by danmois** [comment]
V2
- introduce error handling for axi_slave_adc_read and axi_slave_adc_write functions

**Review by amiclaus — APPROVED** [review]

**Pull request merged**